File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.cpp
class Solution {
public:
    int secondHighest(string s) {
        int large = -1;
        int small = -1;
        for(char i : s){
            if(isdigit(i)){
                int digit = i - '0';
                if(digit > large){
                    small = large;
                    large = digit;
                }
                else if(digit < large && digit > small){
                    small = digit;
                }
            }
        }
        return small;
    }
};

File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.java
class Solution {
    public int secondHighest(String s) {
        int large = -1;
        int second = -1;
        for(char c : s.toCharArray()){
            if(Character.isDigit(c)){
                int i = c -'0';
                if(large < i){
                second =  large;
                large = i;
            }else if(second < i && large > i) second = i;
            }
        }
   
        return second;
    }
}